flink-dstl/flink-dstl-dfs/src/test/java/org/apache/flink/changelog/fs/TestingStateChangeUploader.java

```diff
@@ -26,22 +27,30 @@
 import java.util.List;
 import java.util.concurrent.CopyOnWriteArrayList;
 
+import static java.util.Collections.emptyMap;
 import static java.util.stream.Collectors.toList;
 
 class TestingStateChangeUploader implements StateChangeUploader {
     private final Collection<StateChangeSet> uploaded = new CopyOnWriteArrayList<>();
-    private final List<UploadTask> tasks = new CopyOnWriteArrayList<>();
+    private final List<UploadTask> tasks;
     private boolean closed;
 
+    TestingStateChangeUploader() {
+        tasks = new CopyOnWriteArrayList<>();
```

Comment:
Maybe not related with this PR, why we must make it as a `CopyOnWriteArrayList`?

Reply:
Depending on the test setup, its `upload` method can be called by a dedicated upload thread, and other accesses are from the main thread (e.g. `BatchingStateChangeUploadSchedulerTest`).